doc/pipeline_spec.md

```diff
@@ -0,0 +1,72 @@
+# Pipeline Specification
+
+## Format
+
+```json
+{
+  "pipeline": {
+    "name": string
+  },
+  "transform": {
+    "image": string,
+    "cmd": [ string ],
+    "stdin": [ string ]
+  },
+  "parallelism": int,
+  "inputs": [
+    {
+      "repo": {
+        "name": string
+      },
+      "reduce": bool
+    }
+  ]
+}
+```
+
+`pipeline.name` is the name of the pipeline that you are creating.  Each pipeline needs to have a unique name.
+
+`transform.image` is the name of the Docker image that your jobs run in.  Currently, this image needs to [inherit from a Pachyderm-provided image known as `job-shim`](https://github.com/pachyderm/pachyderm/blob/fae98e54af0d6932e258e4b0df4ea784414c921e/examples/fruit_stand/Dockerfile#L1).
+
+`transform.cmd` is the command passed to the Docker run invocation.  Note that as with Docker cmd is not run inside a shell that means that things like wildcard globbing (`*`), pipes (`|`) and file redirects (`>` and `>>`) will not work.  To get that behavior you should have your command be a shell of your choice (e.g. `sh`) and pass a shell script to stdin.
+
+`transform.stdin` is an array of lines that are sent to your command on stdin.  Lines need not end in newline characters.
+
+`parallelism` is how many copies of your container should run in parallel.  You may pass 0 for parallelism in which case PPS will set the parallelism based on available resources.
+
+`inputs` specifies a set of Repos that will be visible to the jobs during runtime. commits to these repos will cause the pipeline to create new jobs to process them.
+
+`inputs.reduce` specifies how a repo will be partitioned among parallel containers.  If set to true, the data will be partitioned by files.  If set to false, the data will be partitioned by blocks.
+
+## Examples
+
+Here is a modified example from the [fruit stand example](../examples/fruit_stand/GUIDE.md).
+
+```json
+{
+  "pipeline": {
+    "name": "filter"
+  },
+  "transform": {
+    "image": "fruit_stand",
+    "cmd": [ "sh" ],
+    "stdin": [
+        "grep apple  /pfs/data/* >/pfs/out/apple",
+        "grep banana /pfs/data/* >/pfs/out/banana",
+        "grep orange /pfs/data/* >/pfs/out/orange"
+    ]
+  },
+  "parallelism": "4",
+  "inputs": [
+    {
+      "repo": {
+        "name": "data"
+      },
+      "reduce": true
```

Comment:
Does it make sense for filter to be a reduce? I feel like this is clearly a map job. I think the average use case is someone using their own container, not bash. Can we make the image and cmd something more demonstrative? The quickstart guide already has the bash shell example covered.